### PR TITLE
[top, clkmgr, rstmgr] Add clk_aon

### PR DESCRIPTION
**Pull request by tjaychen**

### CODE REVIEW

**Comment by eunchan** [comment]
Hey Tim, do you want this to be merged before branching out to bronze?

**Comment by tjaychen** [comment]
> Hey Tim, do you want this to be merged before branching out to bronze?

that would be really nice :) because then i think we can set proper constraints. What do you think?

**Comment by eunchan** [comment]
Yeah sounds good to me. Let's target this PR to be merged tomorrow as the change is quite straight forward.

File: util/topgen/merge.py [hunk 1]
```diff
@@ -499,12 +499,12 @@ def amend_resets(top):
             log.error("{} missing type field".format(reset["name"]))
             return
 
-        if reset["type"] in ["gen"]:
+        if reset["type"] in ["top"]:
```
Comment (eunchan) [inline util/topgen/merge.py:502]: Do you plan to list the types here? Or always checking one "top"?

Comment (tjaychen) [reply util/topgen/merge.py:502]: ah do you mean i should make the types to check for a separate variable that i check against? I think that might be cleaner.

Comment (eunchan) [reply util/topgen/merge.py:502]: Nope I meant, as the list has only one entry inside, i think this is basically `reset["type"] == "top"`. So I asked if you want to something like this later in your mind. `reset["type"] in ["top", "gen"]` etc

Comment (tjaychen) [reply util/topgen/merge.py:502]: o i see you're right. Originally i had more than one type, but it is one now. Let me update.

**Comment by tjaychen** [comment]
merging, thanks all.

**Pull request merged**